### PR TITLE
Add machine accounts

### LINKED ISSUE #1702: Provide the ability for Team managers to manage (create/delete/modify passwords)  machine accounts via the UI/api

### PR DESCRIPTION
**Pull request by barreiro**
Adds a `machine` user role. 

There is a bit of mismatch as it's presented in the UI as a team role, so that it can be managed by team managers. 

Fixes #1702 .

### CODE REVIEW

**Comment by barreiro** [comment]
added test case

**Comment by johnaohara** [comment]
This PR makes changes to the UI, please can we add/modify the docs?

**Comment by barreiro** [comment]
Added section to the docs with a couple screenshots :grinning:

**Review by johnaohara — CHANGES_REQUESTED** [review]

**Comment by johnaohara** [comment]
> There is a bit of mismatch as it's presented in the UI as a team role, so that it can be managed by team managers.

I think this confuses the role logic and UI slightly, in that "Machine Account" is a "User Role", whereas we are presenting is as a "Team Role", which;

1. Means that machine accounts can be re-used / managed by multiple teams - i.e. think password reset
2. Couples the logic between user roles & team roles.

I think the logic would be simpler, we could guard against re-use and the UI would be more intuitive if under the "Machine Accounts" Tab, we have a "New Account" button that created a new "Machine Account" for a particular team

File: docs/site/content/en/docs/Concepts/core-concepts/users/index.md [hunk 1]
```diff
@@ -36,3 +36,14 @@ Users who explore runs, create and modify new tests should have the `engineers-t
 You can also create a role that allows read-only access to team's private runs, `engineers-viewer` consisting of `engineers-team` and `viewer`.
 
 The `admin` role is not tied to any of the teams.
+                                                    
+
+## Bot accounts (CI)
```
[the remark this answers is not in the document]
Comment (barreiro) [reply docs/site/content/en/docs/Concepts/core-concepts/users/index.md:41]: :+1:

File: docs/site/content/en/docs/Concepts/core-concepts/users/index.md [hunk 2]
```diff
@@ -36,3 +36,14 @@ Users who explore runs, create and modify new tests should have the `engineers-t
...
+
+A team manager can assign the `Machine Account` role to users in his managed team. It allows those accounts to perform HTTP Basic authentication instead of having to authenticate with Keycloak. To generate a password access the `Machine accounts` tabs on the user profile, select the managed team and a list of users with `Machine Account` role appears. 
+
+{{% imgproc machine_accounts_tab Fit "1200x900" /%}}
+
+Click on the red `Reset Password` button and confirm. A dialog with the new generated password appears. Copy the password and close the dilog. This is the only time the password is visible.
```
[the remark this answers is not in the document]
Comment (barreiro) [reply docs/site/content/en/docs/Concepts/core-concepts/users/index.md:47]: :+1:

**Comment by barreiro** [comment]
I have revamped this PR after your comments, with the goal of improving UX.

1. Accounts are created and managed in the `Machine Accounts` tab and in there only.
2. They are not displayed in the UI as users and their roles are not mutable.
3. Because of the way they are created and managed, they are only associated with a single team.
4. Same behavior for both keycloak and database backends, meaning that machine accounts can be consistently created in different scenarios (keycloak, database or external OIDC provider).
5. Test cases now cover both backends, and ensure that these accounts can log in with Basic authentication in both cases.

When migrating between versions, this feature requires that a new `machine` base role is created in keycloak.

**Comment by johnaohara** [comment]
@barreiro this looks good, final question, how do users remove team accounts?

**Pull request merged**